packages/openneuro-app/src/scripts/users/user-datasets-view.tsx

```diff
@@ -7,47 +11,148 @@ interface User {
 interface Dataset {
   id: string
   created: string
-  ownerId: string
   name: string
-  type: string
+  public: boolean
+  analytics: {
+    views: number
+    downloads: number
+  }
+  stars: [{ userId: string; datasetId: string }]
+  followers: [{ userId: string; datasetId: string }]
+  latestSnapshot?: {
+    id: string
+    size: number
+    issues: [{ severity: string }]
+    created?: string
+  }
 }
 
 interface UserDatasetsViewProps {
   user: User
+  hasEdit: boolean
 }
 
-const dummyDatasets: Dataset[] = [
-  {
-    id: "ds00001",
-    created: "2023-11-01T12:00:00Z",
-    ownerId: "1",
-    name: "Dataset 1",
-    type: "public",
-  },
-  {
-    id: "ds00002",
-    created: "2023-11-02T12:00:00Z",
-    ownerId: "2",
-    name: "Dataset 2",
-    type: "private",
-  },
-]
+const DATASETS_QUERY = gql`
+  query GetDatasets($first: Int) {
+    datasets(first: $first) {
+      edges {
+        node {
+          id
+          created
+          name
+          public
+          analytics {
+            views
+            downloads
+          }
+          stars {
+            userId
+            datasetId
+          }
+          followers {
+            userId
+            datasetId
+          }
+          latestSnapshot {
+            id
+            size
+            created
+            issues {
+              severity
+            }
+            description {
+              Authors
+            }
+          }
+        }
+      }
+    }
+  }
+`
```

Comment:
Can this reuse any of the query fragments we already have defined?

https://github.com/OpenNeuroOrg/openneuro/blob/977d5a1facbc48bc9a8af027a1511f6a1517c425/packages/openneuro-app/src/scripts/datalad/dataset/dataset-query-fragments.js

Reply:
I moved this into a new Fragment I did not find one that I could reuse.